pkg/frontend/transport/handler.go

```diff
@@ -117,6 +120,11 @@ func NewHandler(cfg HandlerConfig, roundTripper http.RoundTripper, log log.Logge
 }
 
 func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	activityIndex := f.at.Insert(func() string {
+		return httpRequestActivity(r)
+	})
+	defer f.at.Delete(activityIndex)
```

Comment:
since `httpRequestActivity` parses request, this needs to happen after we set the correct limit on the body later in this method.

Reply:
Done after rebasing.